.github/workflows/kotsadm.yaml

```diff
@@ -0,0 +1,565 @@
+        GIT_TAG: ${{ steps.get_tag.outputs.GIT_TAG }}
+        # GITHUB_REPOSITORY:
+        # GITHUB_WORKSPACE:
+        SCOPE_DSN_PUBLIC: ""
+      run: SCOPE_DSN=$SCOPE_DSN_PUBLIC make -C kotsadm test kotsadm
+
+    - name: Upload Go API artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: go_api
+        path: ./kotsadm/bin
+
+
+  build_typescript_api:
+    runs-on: ubuntu-18.04
+    needs: [test_web, build_web, build_operator, build_kurl_proxy]
+    container:
+      image: replicated/gitops-builder:buildkite
+      options: --user root
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Get the version
+      if: startsWith(github.ref, 'refs/tags/')
+      id: get_tag
+      run: echo ::set-output name=GIT_TAG::${GITHUB_REF/refs\/tags\//}
+
+    - name: Build TypeScript API
+      env:
+        GIT_COMMIT: ${{ github.sha }}
+        GIT_TAG: ${{ steps.get_tag.outputs.GIT_TAG }}
+        SCOPE_DSN_PUBLIC: ""
+        AUTO_CREATE_CLUSTER_TOKEN: this-is-a-test-value
+      run: make -C kotsadm/api no-yarn deps build
+
+    - name: Upload Go API build artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: typescript_api_build
+        path: ./kotsadm/api/build
+
+    - name: Upload Go API bin artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: typescript_api_bin
+        path: ./kotsadm/api/bin
+
+
+  release_go_api_alpha:
+    runs-on: ubuntu-18.04
+    if: ${{ github.ref == 'refs/heads/master' }}
+    needs: [build_web, build_go_api]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Setup Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: "1.14.3"
+
+    - name: Download go_api artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: go_api
+        path: ./kotsadm/bin
+
+    - name: Download web artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: web
+        path: ./kotsadm/web/dist
+
+    - name: Build alpha release
+      env:
+        GIT_COMMIT: ${{ github.sha }}
+      run: |
+        docker login -u ${{ secrets.DOCKERHUB_USER }} -p ${{ secrets.DOCKERHUB_PASSWORD }}
+        make -C kotsadm build-alpha
+
+
+  release_go_api_tagged:
+    runs-on: ubuntu-18.04
+    needs: [build_web, build_go_api]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Setup Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: "1.14.3"
+
+    - name: Get the version
+      if: startsWith(github.ref, 'refs/tags/')
+      id: get_tag
+      run: echo ::set-output name=GIT_TAG::${GITHUB_REF/refs\/tags\//}
+  
+    - name: Download go_api artifact
+      if: ${{ steps.get_tag.outputs.GIT_TAG != '' }}
+      uses: actions/download-artifact@v2
+      with:
+        name: go_api
+        path: ./kotsadm/bin
+
+    - name: Download web artifact
+      if: ${{ steps.get_tag.outputs.GIT_TAG != '' }}
```

Comment:
is there a way to put the 'if' a level higher?

Reply:
no, because github has no global level tag variable.  it has to be generated as a step inside each job

Reply:
job level condition can be controlled with `on.push.tags`, but that applies to the entire workflow and then we'd have to have separate workflows for tagged and untagged steps, and there would be a lot of duplication.